test/main.js

```diff
@@ -5,55 +5,55 @@ const fs = require('fs');
 
 const noop = () => {};
 
+// Load and override configuration before starting the server
+let config;
+try {
+	require.resolve('../config/config');
+} catch (err) {
+	if (err.code !== 'MODULE_NOT_FOUND' && err.code !== 'ENOENT') throw err; // Should never happen
+
+	console.log("config.js doesn't exist - creating one with default settings...");
+	fs.writeFileSync(
+		path.resolve(__dirname, '../config/config.js'),
+		fs.readFileSync(path.resolve(__dirname, '../config/config-example.js'))
+	);
+} finally {
+	config = require('../config/config');
+}
+Object.assign(config, require('../config/config-example'));
+
+// Actually crash if we crash
+config.crashguard = false;
+// Don't allow config to be overridden while test is running
+config.watchconfig = false;
+// Don't try to write to file system
+config.nofswriting = true;
+// Test a normal ladder
+config.fakeladder = false;
+
+require('./../.lib-dist/process-manager').disabled = true;
+
+// stop chatrooms from loading through modifying the require cache
+try {
+	const chatrooms = require('../config/chatrooms.json');
+	chatrooms.splice(0, chatrooms.length);
+} catch (e) {}
+
+// Don't create a REPL
+require('../.lib-dist/repl').Repl.start = noop;
+
+// Start the server.
+// NOTE: This used "server" before when we needed ".server-dist"
+require('../.server-dist');
```

Comment:
This line, for example, sets up the `Rooms` global so it's accessible at the top level of test files.

Reply:
As far as I can tell, you just moved this up from line 54.

Reply:
Yes; prior to my change, it was in a `before()` function, so `Rooms` wouldn't be accessible at the top level of test files (only within `before`/`after`/`it`/other Mocha functions).

Reply:
Oh! That makes sense.